### PR TITLE
match now resets when needs to and reset to first empty match of team…

### PR DESCRIPTION
**Pull request by idanstark42**
… after changing it

### CODE REVIEW

File: client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js [hunk 1]
```diff
@@ -146,17 +147,22 @@ class MetadataInputsController {
...
+      }
+      if (this.matches.some(match => this.data.current.stage === match.stage || this.data.current.round === match.round)) {
+        return Promise.resolve()
+      }
+    }
+    this.forceMatchReselection = false
+    const firstIncompleteMatch = this.matches.find(match => !match.complete)
+    if (firstIncompleteMatch) {
+      this.data.current.stage = firstIncompleteMatch.stage
+      this.data.current.round = firstIncompleteMatch.round
```
Comment (2roy999) [inline client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js]: this function doesn't return promise in all of it's paths

Comment (alanggreen) [reply client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js]: @2roy999 did you mean to approve the PR? This sounds important to me

Comment (idanstark42) [reply client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js]: It's not really important in the way it's used now, but I fixed it anyway

Comment (idanstark42) [reply client/components/scoresheet/metadata_inputs/metadata_inputs_controller.js]: It's not really important in the way it's used now, but I fixed it anyway

**Pull request merged**